code/ship/ship.cpp

```diff
@@ -9399,6 +9401,9 @@ int ship_create(matrix* orient, vec3d* pos, int ship_type, const char* ship_name
 		strncpy(shipp->ship_name, Ship_info[ship_type].name, name_len);
 		strcpy(shipp->ship_name + name_len, suffix);
 	} else {
+		if (ship_find_exited_ship_by_name(ship_name) >= 0 && !(Game_mode & GM_MULTIPLAYER)) {
+			Warning(LOCATION, "A newly-arrived ship has been given the same name as a ship previously destroyed in-mission. This can cause unpredictable SEXP behavior. Correct your mission file or scripts to prevent duplicates.");
+		}
```

Comment:
The warning ought to display what the specified name was and what the replacement default name will be.

Reply:
I'm adding ship_name, but in this case the name would not be replaced.  I can add the default name of what it *would* be if we were to replace it with the default name.

Reply:
Oh, right.  Yes.